tests/datasource/gwas_catalog/test_gwas_catalog_curation.py

```diff
@@ -0,0 +1,252 @@
+        """Generate a dataset with a given set of studies with minimalistic schema."""
+        # Generating a study index with 7 studies where the first study (s0) has no summary statistics:
+        study_data = [
+            (f"s{count}", False) if count == 0 else (f"s{count}", True)
+            for count in range(0, 7)
+        ]
+        columns = [
+            "projectId",
+            "studyType",
+            "traitFromSource",
+            "publicationTitle",
+            "publicationFirstAuthor",
+        ]
+        return StudyIndexGWASCatalog(
+            _df=(
+                spark.createDataFrame(study_data, ["studyId", "hasSumstats"]).select(
+                    "*", *[f.lit("foo").alias(colname) for colname in columns]
+                )
+            ),
+            _schema=StudyIndexGWASCatalog.get_schema(),
+        )
+
+    @pytest.fixture(scope="class")
+    def mock_study_curation(
+        self: TestGWASCatalogStudyCuration, spark: SparkSession
+    ) -> DataFrame:
+        """Generate a mocked curation dataset with matching studies."""
+        curation_data = [
+            ("s2", None, None, None, True),  # Good study
+            ("s3", "pQTL", None, None, True),  # Update type
+            ("s4", None, "analysis 1", None, True),  # Add analysis flag
+            ("s5", None, None, "QC flag", True),  # Add analysis flag
+        ]
+
+        curation_columns = [
+            "studyId",
+            "updateStudyType",
+            "upateAnalysisFlags",
+            "upateQualityControls",
+            "isCurated",
+        ]
+        return spark.createDataFrame(curation_data, curation_columns)
+
+    @staticmethod
+    def test_curation__return_type(
+        mock_gwas_study_index: StudyIndexGWASCatalog, mock_study_curation: DataFrame
+    ) -> None:
+        """Test return types."""
+        # Method should work on empty curation:
+        assert isinstance(
+            mock_gwas_study_index.annotate_from_study_curation(None),
+            StudyIndexGWASCatalog,
+        ), f"When applied None to curation function the returned type was: {type(mock_gwas_study_index.annotate_from_study_curation(None))}"
+
+        # Return type should work:
+        assert isinstance(
+            mock_gwas_study_index.annotate_from_study_curation(mock_study_curation),
+            StudyIndexGWASCatalog,
+        ), f"When applied None to curation function the returned type was: {type(mock_gwas_study_index.annotate_from_study_curation(mock_study_curation))}"
+
+    @staticmethod
+    def test_curation__returned_rows(
+        mock_gwas_study_index: StudyIndexGWASCatalog, mock_study_curation: DataFrame
+    ) -> None:
+        """Test return types."""
+        expected_count = mock_gwas_study_index.df.count()
+        zero_return_count = mock_gwas_study_index.annotate_from_study_curation(
+            None
+        ).df.count()
+        return_count = mock_gwas_study_index.annotate_from_study_curation(
```

Comment:
`return_count` and `zero_return_count` are the same. is this intended?

Reply:
The `zero_return_count` asserts that the number of returned studies won't change even if there's no curation table provided for the curation funcion. Might be an overly cautious test, but that's why it's tested under the same funcition.